### ISSUE TITLE
Support for definining storageClassNames

### ISSUE THREAD

**Issue by MisterTimn** [streamnative/charts#126]
We've been switching over from a custom pulsar helm deployment to using this one.

I've noticed that the PVC template functions either use a local-storage class or define a new one based on the `storageClass` object within the different components.

In our on-prem k8s cluster we have already defined several storageclasses, so we would only need a place to add our `storageClassName` so an appropriate PVC is made.

e.g.

```yaml
zookeeper:
  component: zookeeper
  replicaCount: 3
  volumes:
    # use a persistent volume or emptyDir
    persistence: true
    data:
      name: data
      size: 50Gi
      local_storage: false
      storageClassName: <our-custom-storageclass>
```

Now we can only deploy relying on our `default` storageclass but this is less then ideal.

Is it enough to add this `storageClassName` entry to all component's volumes object and add a clause to the template functions which make the PVCs? I could add this and start a pull request.

**Comment by sijie** [comment]
@MisterTimn Yes! That sounds like a good idea. Would you like to make a contribution?

**Comment by MisterTimn** [comment]
First thing I'm noticing is how the config of these storageclass has been spread throughout the different components and everytime it has been tackled in a different way.

In prometheus its handled inside the pvc-template
In zookeeper and broker its handled in the helpers but each in its own way (print functions vs full include defines.

I'm working on just adding the storageClassName now but maybe we should move towards a unified and more concise way of handling this? Define these classes in the same way across all components.

I have always enjoyed bitnami charts and how they do the defining of storage classes
Reference:
https://github.com/bitnami/charts/blob/007da4ff5a6b80968525a324219645c668cd6fed/bitnami/influxdb/templates/_helpers.tpl#L270
https://github.com/bitnami/charts/blob/007da4ff5a6b80968525a324219645c668cd6fed/bitnami/influxdb/values.yaml#L529

If we would adapt such a style this would mean quite a big overhaul.

**Comment by sijie** [comment]
@MisterTimn Looks good to me! Let's move towards that direction.

**Comment by MisterTimn** [comment]
Ive made a PR with just the addition of the `storageClassName` values for now, don't have any spare time at the moment to overhaul/streamline volume config throughout the chart.

**Comment by sijie** [comment]
@MisterTimn Thank you!

**Comment by sijie** [comment]
This is fixed by #127
